EIPS/eip-7525.md

```diff
@@ -0,0 +1,161 @@
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-25
+---
+
+## Abstract
+
+In the Execution Layer, accounts that have been inactive for a certain period are expired. The expired state can be recovered through a special transaction for restoration. This significantly lowers the entry barrier for node operation by maintaining only the minimal required state for validation.
+
+## Motivation
+
+Numerous attempts to create a stateless roadmap exist, but we still lack practical solutions, and the state size keeps increasing quickly. Recently, researchers at a16z pointed out the significant challenges in maintaining a stateless chain, due to the heavy load of witness updates. Given these hurdles, we believe that developing mechanisms for state expiry is crucial for the community.
+
+According to our analysis, we have observed that 95% of the accounts in Execution Layer have been inactive for at least one month. The inactive accounts refer to those that have not been in engaged or affected by any transactions.
+
+In other words, this means that only a mere 5% of the data is utilized when validating the data in the current Execution Layer. In most situations, historical data is not essential and just serves to significantly burden the system.
+
+This EIP enhances node efficiency by keeping data only for active accounts needed for data verification. This approach optimizes node operations and reduces the burden of managing inactive accounts by expiring them periodically.
+
+## Specification
+
+### State Data Management
+
+The data that was previously managed in a single database is divided and managed separately for accounts and the rest. To facilitate ease of deletion, trie data structure is stored in the database, separated for each epoch.
+
+If data for a certain epoch is not needed, instead of going through the process of traversing the trie, you can simply delete the file corresponding to that epoch.
+
+### Sweep
+
+A config called `SWEEP_EPOCH` is added to specify how often to expire inactive accounts. `SWEEP_EPOCH` refers to the period of performing sweeps, and sweeps are executed every epoch corresponding to the value specified here.
+
+The state trie stores the activities of each account every epoch. At each checkpoint block, the previously created trie become closable and then it starts from an empty trie. We call the frozen trie a checkpoint trie.
+
+During the subsequent epochs, when the state of any account needs to be updated, if the account is not in the current trie, we perform a process to find information about the account from the previous checkpoint trie and add it to the current trie. If the account is already in the current trie, the update is performed immediately.
+
+In this process, the account-related state is stored in the DB handling the account data mentioned earlier. The sweep is only conducted for account-related data, and does not apply to things like storage data.
```

Comment:
you mean the storage state is not expired?

Reply:
We can simply expire the storage state, but I haven't determined the best approach for the restoration process of the storage slots. However, as there may be room for misunderstanding, I will consider making revisions to the content.

Reply:
Without robust support for expiring and reviving storage data this EIP is not very useful.